src/middleware/dataview.middleware.js

```diff
@@ -34,8 +34,19 @@ export const setRecordCount = (req, res, next) => {
 export const constructTableParams = (req, res, next) => {
   const { entities, uniqueDatasetFields } = req
 
-  const columns = uniqueDatasetFields
-  const fields = uniqueDatasetFields
+  const leadingFields = []
+  for (const field of uniqueDatasetFields) {
+    if (field === 'reference') {
+      leadingFields.splice(0, 0, field)
+    } else if (field === 'name') {
+      leadingFields.push(field)
+    }
+    if (leadingFields.length === 2) break
+  }
+  const orderedDatasetFields = [...leadingFields, ...uniqueDatasetFields.filter(field => !leadingFields.includes(field))]
+
+  const columns = orderedDatasetFields
+  const fields = orderedDatasetFields
```

Comment:
This might be more cleaner

```suggestion
  const leadingFields = uniqueDatasetFields.filter(field => 
    field === 'reference' || field === 'name'
  ).sort(field => (field === 'reference' ? -1 : 1));

  const orderedDatasetFields = [
    ...leadingFields,
    ...uniqueDatasetFields.filter(field => !leadingFields.includes(field))
  ];

  const columns = orderedDatasetFields;
  const fields = orderedDatasetFields;
```

Reply:
Hm, both of the above do not bring joy. The fields array does not need to be iterated over more than once.